posts/inside-rust/2022-08-11-diagnostic-effort.md

```diff
@@ -0,0 +1,336 @@
+---
+layout: post
+title: "Contribute to the diagnostic translation effort!"
+author: David Wood
+team: the compiler team <https://www.rust-lang.org/governance/teams/compiler>
+---
+# Contribute to the Diagnostic Translation Effort!
+The Rust Diagnostics working group is leading an effort to add support
+internationalization of error messages in the compiler, allowing the compiler
+to produce output in languages other than English.
+
+For example, consider the following diagnostic where a user has used a colon to
+specify a function's return type instead of an arrow:
+
+```
+error: return types are denoted using `->`
+ --> src/main.rs:1:21
+  |
+1 | fn meaning_of_life(): u32 { 42 }
+  |                     ^ help: use `->` instead
+```
+
+We could output that diagnostic in Chinese:
+
+```
+错误: 返回类型使用`->`表示
+ --> src/main.rs:1:21
+  |
+1 | fn meaning_of_life(): u32 { 42 }
+  |                     ^ 帮助: 使用`->`来代替
```

Comment:
Before we post this, we should make sure our encoding headers are set up properly for this to render right on the site (not just in GitHub). You can build the site with cargo run and that should give you some links to click which are a good enough start, I think.

Reply:
![image](https://user-images.githubusercontent.com/1295100/184386371-97ce7ab1-1fd3-478d-9649-fa65cbad53eb.png)
Seems to work fine for me in Chrome :)

Reply:
I've also added `text` to those code snippets so they don't get syntax highlighted incorrectly.

Reply:
For Chinese, Japanese and Korean to be rendered correctly I believe you have to set the `lang` html attribute to the right value. Unicode decided to merge characters from those languages together if they mean the same, even if they need to be rendered differently, so the only way to know how to render them is setting the right `lang` attribute value as far as I understand it.

Reply:
I've used html tags instead of Markdown code blocks for the translated diagnostic examples, so that I can add the `lang` attribute to them. It makes the Markdown a little bit uglier but it seems unavoidable.

Reply:
Breaking the lines where I have is unfortunately required to get it to look the way that it did before:

![image](https://user-images.githubusercontent.com/1295100/184878981-e4ad1781-fef8-45e2-baab-c844222584c7.png)